docs/mobile-apps/automated-testing/flutter/flutter-integration-testing-android.md

```diff
@@ -0,0 +1,217 @@
+---
+id: flutter-integration-testing-android
+title: Flutter Android
+sidebar_label: Flutter Android
+description: Run your Flutter integration tests for Android
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+Integration testing involves testing various modules of the software under development as a group to see whether they function together seamlessly.
+Ahead, we’ll take a closer look at how flutter supports the integration testing for Android.
+
+Follow this guide to run integration tests for your Flutter app on Android.
+
+
+## What You'll Need
+
+- A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up))
+- Your Sauce Labs [Username and Access Key](https://app.saucelabs.com/user-settings)
+- A mobile app file If you don't have one, consider using our flutter Demo Apps:
+    - [Sauce Labs Flutter Demo App](https://github.com/saucelabs/my-demo-app-flutter)
+
+## Prepare Your Flutter Application For Integration Testing
+
+1. Open your Flutter project in your favorite IDE.
+2. In your Flutter app's `pubspec.yaml`, add the following dependencies:
```

Comment:
we need to add a 1 liner here, that they need to compile an Espresso testapp and upload that too to the platform

Reply:
Your Espresso apps and files for real devices and

Reply:
> 12:24:52 INF Checking if ...../my-demo-app-flutter/build/app/outputs/flutter-apk/app-debug.apk has already been uploaded previously
> 12:24:52 INF Checksum: 1df0b6684973536fef4ae653d89661d6c48d6f699091511515b69735d6a80fbd
> 12:27:26 INF Application uploaded. durationMs=153173 storageId=6849a64a-3c51-4423-87f2-b3660c972a36
> 12:27:26 INF Checking if ...../build/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk has already been uploaded previously
> 12:27:26 INF Checksum: b5b15cb741b78fe7a5df171406c17ec9ea8fd6ac52623abf7a8df519270e281d